### PR TITLE
allow "host" to be specified for HostPlatform()

### PR DESCRIPTION
**Pull request by sjkelly**
This is useful when locally developing and testing _jll files
where there may be more specificity than the triplet.
This should allow for:

```
julia build_tarballs.jl --deploy=local host
```

as a new idiom for ease of local iteration.

### CODE REVIEW

**Comment by giordano** [comment]
Interesting idea! This needs to be tested though

File: test/platforms.jl [hunk 1]
```diff
@@ -111,3 +111,11 @@ using BinaryBuilderBase: get_march_flags, get_all_arch_names, get_all_march_name
...
     @test sort(get_all_march_names("armv7l")) == ["armv7l", "neonvfpv4"]
 end
+
+@testset "parse" begin
+    @test parse_platform("host") == HostPlatform()
+    @test parse_platform("any") == AnyPlatform()
+    @test parse_platform("x86_64-linux") == Platform("x86_64", "linux")
+    @test parse_platform("x86_64-linux-musl") == Platform("x86_64", "linux", libc="musl")
+    @test parse_platform("x86_64-linux-musl-gcc7") == Platform("x86_64", "linux", libc="musl", libgfortran_version=v"4.0.0")
+end
```
Comment (giordano) [inline test/platforms.jl:121]:
```suggestion
end

```

File: test/platforms.jl [hunk 2]
```diff
@@ -111,3 +111,11 @@ using BinaryBuilderBase: get_march_flags, get_all_arch_names, get_all_march_name
     @test sort(get_all_march_names("x86_64")) == ["avx", "avx2", "avx512", "x86_64"]
     @test sort(get_all_march_names("armv7l")) == ["armv7l", "neonvfpv4"]
 end
+
+@testset "parse" begin
+    @test parse_platform("host") == HostPlatform()
+    @test parse_platform("any") == AnyPlatform()
+    @test parse_platform("x86_64-linux") == Platform("x86_64", "linux")
+    @test parse_platform("x86_64-linux-musl") == Platform("x86_64", "linux", libc="musl")
+    @test parse_platform("x86_64-linux-musl-gcc7") == Platform("x86_64", "linux", libc="musl", libgfortran_version=v"4.0.0")
```
Comment (giordano) [inline test/platforms.jl:120]: We shouldn't really be using the gcc stuff like that nowadays, I'm even surprised that works at all

Comment (sjkelly) [reply test/platforms.jl:120]: TBH this was Copilot's idea, with some proofing :P. I guess the AI is pretty good at these things.

Comment (staticfloat) [reply test/platforms.jl:120]: That's mildly terrifying

Comment (giordano) [inline test/platforms.jl:120]: Let's do some more varied and useful tests:
```suggestion
    @test parse_platform("i686-w64-mingw32-cxx03") == Platform("i686", "windows"; cxxstring_abi="cxx03")
    @test parse_platform("aarch64-apple-darwin21-libgfortran4-cxx11") == Platform("aarch64", "macos"; os_version=v"21", cxxstring_abi="cxx11", libgfortran_version=v"4")
```

**Review by giordano — APPROVED** [review]
Let's override the machines.

Unrelated to this PR: tests on julia nightly are failing again (and I'm not going to look into that).  Sigh.

**Pull request merged**